src/primitive/polyhedron.rs

```diff
@@ -0,0 +1,713 @@
+        let mut face = Face {
+            edge: 0,
+            vertices: (a, b, c),
+            plane: Plane::from_points(
+                vertices[a].position,
+                vertices[b].position,
+                vertices[c].position,
+            ).unwrap(),
+            ready: false,
+        };
+        println!("{:?}", face);
+        let face_index = faces.len();
+        let mut face_edge_indices = [0, 0, 0];
+        for j in 0..3 {
+            let i = if j == 0 { 2 } else { j - 1 };
+            let v0 = face_vertices[i];
+            let v1 = face_vertices[j];
+
+            let (edge_v0_v1_index, edge_v1_v0_index) = if let Some(edge) = edge_map.get(&(v0, v1)) {
+                (edge.clone(), edges[*edge].twin_edge)
+            } else {
+                let edge_v0_v1_index = edges.len();
+                let edge_v1_v0_index = edges.len() + 1;
+
+                edges.push(Edge {
+                    target_vertex: v1,
+                    left_face: 0,
+                    next_edge: 0,
+                    previous_edge: 0,
+                    twin_edge: edge_v1_v0_index,
+                    ready: false,
+                });
+
+                edges.push(Edge {
+                    target_vertex: v0,
+                    left_face: 0,
+                    next_edge: 0,
+                    previous_edge: 0,
+                    twin_edge: edge_v0_v1_index,
+                    ready: false,
+                });
+
+                (edge_v0_v1_index, edge_v1_v0_index)
+            };
+
+            edge_map.insert((v0, v1), edge_v0_v1_index);
+            edge_map.insert((v1, v0), edge_v1_v0_index);
+
+            if !edges[edge_v0_v1_index].ready {
+                edges[edge_v0_v1_index].left_face = face_index;
+                edges[edge_v0_v1_index].ready = true;
+            }
+
+            if !vertices[v0].ready {
+                vertices[v0].edge = edge_v0_v1_index;
+                vertices[v0].ready = true;
+            }
+
+            if !face.ready {
+                face.edge = edge_v0_v1_index;
+                face.ready = true;
+            }
+
+            face_edge_indices[i] = edge_v0_v1_index;
+        }
+
+        faces.push(face);
+
+        for j in 0..3 {
+            let i = if j == 0 { 2 } else { j - 1 };
+            let edge_i = face_edge_indices[i];
+            let edge_j = face_edge_indices[j];
+            edges[edge_i].next_edge = edge_j;
+            edges[edge_j].previous_edge = edge_i;
+        }
+    }
+
+    (vertices, edges, faces)
+}
+
+impl<S> SupportFunction for ConvexPolyhedron<S>
+where
+    S: BaseFloat,
+{
+    type Point = Point3<S>;
+
+    fn support_point<T>(&self, direction: &Vector3<S>, transform: &T) -> Point3<S>
+    where
+        T: Transform<Point3<S>>,
+    {
+        let p = match self.mode {
+            PolyhedronMode::VertexOnly => {
+                self.brute_force_support_point(
+                    transform
+                        .inverse_transform()
```

Comment:
If this is so often needed, we might want to add `inverse_transform_vector` to cgmath having a default implementation (so that it's not a breaking change). Most implementors should be able to do this more efficiently.

Reply:
Yeah, that would be nice. I'll see if I can get a PR in there later. It's something that's immensely useful for collision detection, as you can see. It's way cheaper to transform a direction vector in the inverse, than transforming all the vertices of a polyhedron etc.

Reply:
And, yes, this is just an inverse rotation and scaling, since VectorSpace has no concept of translation.

Reply:
https://github.com/brendanzab/cgmath/pull/431